qcodes/plots/qcmatplotlib.py

```diff
@@ -41,6 +43,33 @@ def align_y_axis(ax, ax_target):
     posn_old, posn_target = ax.get_position(), ax_target.get_position()
     ax.set_position([posn_old.x0, posn_target.y0, posn_old.width, posn_target.height])
 
+def set_zscale(self, scale : str):
+    if not hasattr(self, 'qcodes_colorbar'):
+        raise RuntimeError("Axes object does not have a colorbar.")
+    data_arrays = [data_array.get_array() for data_array in self.collections]
+    clim = [
+        np.min([np.nanmin(data_array) for data_array in data_arrays]),
```

Comment:
Might be worthwhile making the outer one nanmin/nanmax as well, in case a dataarray has only nan elements

Reply:
Done.